### DISCUSSION TITLE
Combine lathe Top slide and cross slide axis

### DISCUSSION THREAD

**Discussion by IanVicAus** [bartei/rotary-controller-python#19] [Ideas]
On a mill you are working to combine the 2 vertical axis.
On a lathe it would be useful to be able to combine the cross and top slide with the top slide being able to be at an angle.
In this way very fine movement can be achieved using the top slide and the position of the tool still known.

**Comment by bartei** [comment]
This should be easily doable for display only, the challenge would be trying to also handle the servo motor in function of the position of the combined axes.

When configuring an angle for the top slide there is trigonometry involved and right now the rotary controller only understands rational numbers (fractions N/D) trigonometric values are in the R domain. Though if approximation is acceptable it might work...

**Reply by IanVicAus** [reply]
There are 2 use cases.
Threading 
As you move down the thread at some angle (30deg or 29deg for a 60deg thread).
Having the movement equivalent to cross slide movement would be advantageous as then you can thread from OD to ID accurately
So this would actually be just trig on the 3rd axis resolving to either cross/axial movement or both?
Process for threading would be a zero point on the cross slide (tool at OD for ext thd) and cutting depth set with top slide.
ELS using the axial position of the saddle as is usual
I don't think there is a worry as far as axial motion is concerned - no one else worries. 

Taper
Having all summed axial from saddle and top slide trig and radial from cross slide and top slide trig would mean that you could follow a taper and reposition the tool using any axis and get to the same point again.
So machine a taper that is longer than the top slide travel by cutting to point - zeroing all axis and then repositioning so that the tool is at the zero point and then continuing the taper

Another use case maybe following a radius path. You could use tool setting attributes to enable a  ball to be turned. Changing the tool type at the max dia.
For this type of turning a color guide may be useful. a diamond indicator with a centre point?
When on the curve the centre point is lit
As you move it fades to indicator colors as you move off the designated path with the error also being indicated by the 4 positions around the centre point.
The way it is done today is creating a step process - resulting in a 3D printing type surface. If a large radius tool (say 8mm dia) is used and a small step size is selected it you can get a reasonable surface finish. It also depends on whether the path is define by a constant angular step or a constant axis movement. Moving to a fully calculated path would allow the operator to more accurately follow the path - with the indicator "lights" providing constant feedback.

Also this is the thoughts of a raving looney from Australia so treat as appropriate.
That said I am trying to think of ways you can make your controller differentiated in the market and with more obvious value to the hobbyist.
